### PR TITLE
linting: clean up doc strings and add editorconfig

### PR DESCRIPTION
**Pull request by seanfarley**
This is mostly just a poke to see if there's any interest in these kinds of
changes :-)

### CODE REVIEW

File: simple-mpc-query.el [hunk 1]
```diff
@@ -5,6 +5,7 @@
 ;; Author: Joren Van Onder <joren.vanonder@gmail.com>
 ;; Maintainer: Joren Van Onder <joren.vanonder@gmail.com>
 ;; Keywords: multimedia, mpd, mpc
+;; Package-Requires: ((emacs "24.4") (s "1.10.0"))
```
Comment (jorenvo) [inline simple-mpc-query.el:8]: Any reason why you added `(emacs "24.4")`? Not saying it's wrong, just not sure why it's needed :).

File: simple-mpc-mode.el [hunk 2]
```diff
@@ -27,6 +27,8 @@
 
 ;;; Code:
 
+(declare-function simple-mpc "simple-mpc")
```
Comment (jorenvo) [inline simple-mpc-mode.el:30]: Why is this needed?

**Review by jorenvo — COMMENTED** [review]
Hi, I think most of this looks fine. Just have a couple of questions :).

By the way, I always wanted to explore and apply some sort of auto formatter. Something similar to [black for Python](https://black.readthedocs.io/en/stable/) or [prettier for JS](https://prettier.io). I know options exist for elisp as well. I'm not sure you've ever looked into these but I'd be open to your thoughts (and/or PRs) on that as well.

File: simple-mpc-query.el (hunk 1 above)
Comment (seanfarley) [reply simple-mpc-query.el:8]: Ah, because I _think_ `s` requires at least emacs 24. Also, because, well, emacs 24 seems to be safe-ish enough to consider "old" :-)

File: simple-mpc-mode.el (hunk 2 above)
Comment (seanfarley) [reply simple-mpc-mode.el:30]: Ah, yes, because further down below (around line 54) is this line:

```lisp
(set (make-local-variable 'revert-buffer-function) #'simple-mpc)
```

and `#'simple-mpc` is not known to be defined without that the `declare-function`. And it can't be fixed with a `require` due to a cyclic dependency on `simple-mpc-mode` in `simple-mpc.el`

**Comment by seanfarley** [comment]
> Hi, I think most of this looks fine. Just have a couple of questions :).
> 
> By the way, I always wanted to explore and apply some sort of auto formatter. Something similar to [black for Python](https://black.readthedocs.io/en/stable/) or [prettier for JS](https://prettier.io). I know options exist for elisp as well. I'm not sure you've ever looked into these but I'd be open to your thoughts (and/or PRs) on that as well.

Ah, I've never used any to be honest! This pullrequest was mostly just me opening up the project with flycheck turned on for elisp and fixing the red lines :-)

**Comment by seanfarley** [comment]
(gentle ping)

**Comment by jorenvo** [comment]
Hi! Sorry, it's taking a while. I haven't merged it because of the `declare-function` you added.

I confirmed it gets rid of a warning during byte-compilation which is great! But it doesn't feel like the right way to solve the issue. I remembered that when I submitted `simple-mpc` to MELPA years ago I had a bunch of them and was [asked to remove them](https://github.com/melpa/melpa/pull/2919#issuecomment-120842078).

Ideally I want to remove the circular dependency so we can do a regular `require`. I'll probably spend a little time on it this weekend. If you have any ideas let me know!

**Comment by seanfarley** [comment]
Ah, yeah, that's a good point. Splitting out the functions to avoid a circular dependency is a much better solution. I'm a bit swamped this weekend else I would investigate it more :-/

**Comment by jorenvo** [comment]
It's merged! I don't have the interest to dive deeply into how to restructure the modes to avoid the circular dependency. Luckily it doesn't cause any real issue, just a warning. So I removed the `declare-function` commit since it doesn't really solve the issue. I also removed the `Package-Requires:` header you added in `simple-mpc-query.el`. When MELPA builds the package it uses only the headers in the main file (`simple-mpc.el`) and it's already in there.

Thanks for your work :).

**Comment by seanfarley** [comment]
Wooo, thanks! If I find the time, I'll also try to take a crack at the circular dependency. Also, I didn't realize that about the MELPA build process; good to know!
